File: src/main/java/com/pduda/field/LocalClient.java
package com.pduda.field;

public class LocalClient {

    public static void main(String[] args) {
        ShoppingCart shoppingCart = new ShoppingCart(10);
        System.out.println("shoppingCart.numberOfProducts() = " + shoppingCart.numberOfProducts());
        System.out.println("shoppingCart.calculateTotalPrice() = " + shoppingCart.calculateTotalPrice());
        System.out.println("shoppingCart.hasDiscount() = " + shoppingCart.hasDiscount());
    }

}

File: Cpp/Source/Field/source/LocalClient.cpp

#include "../include/ShoppingCart.h"

#include <iostream>

int main(int argc, char** argv) 
{
    ShoppingCart shoppingCart;
    shoppingCart.add(10);
    std::cout 
        << "shoppingCart.numberOfProducts() = " << shoppingCart.numberOfProducts() << '\n'
        << "shoppingCart.calculateTotalPrice() = " << shoppingCart.calculateTotalPrice() << '\n'
        << "shoppingCart.hasDiscount() = " << shoppingCart.hasDiscount();
        
    return 0;
}
